File: official/docs/ruby/current/reports/list.rb
require 'easypost'

client = EasyPost::Client.new(api_key: ENV['EASYPOST_API_KEY'])

reports = client.report.all(
  # Replace `payment_log` with any of the report types listed above
  type: 'payment_log',
  page_size: 5,
)

puts reports

File: official/docs/csharp/current/reports/list.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Newtonsoft.Json;
using EasyPost;
using EasyPost.Models.API;
using EasyPost.Parameters;

namespace EasyPostExamples
{
    public class Examples
    {
        public static async Task Main()
        {
            var client = new EasyPost.Client("EASYPOST_API_KEY");

            Parameters.Report.All parameters = new()
            {
                Type = "payment_log",
                PageSize = 5,
                StartDatetime = "2022-10-01"
            };

            ReportCollection reportCollection = await client.Report.All(parameters);

            Console.WriteLine(JsonConvert.SerializeObject(reportCollection, Formatting.Indented));
        }
    }
}
